### PR TITLE
schedulers: avoid recalculate tolerantRatio

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve?
Improve `GetStoreRegionCount`
Part of https://github.com/tikv/pd/issues/3744

### What is changed and how it works?
Calculate `tolerantSizeRatio` when the `balancePlan` is created and reuse it later.

### Check List

Tests
- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: server/schedulers/utils.go [hunk 1]
```diff
@@ -119,24 +121,24 @@ func (p *balancePlan) shouldBalance(scheduleName string) bool {
...
+		return int64(p.tolerantSizeRatio)
 	}
-
 	regionSize := p.region.GetApproximateSize()
 	if regionSize < p.cluster.GetAverageRegionSize() {
 		regionSize = p.cluster.GetAverageRegionSize()
 	}
-	regionSize = int64(float64(regionSize) * adjustTolerantRatio(p.cluster))
-	return regionSize
+	return int64(float64(regionSize) * p.tolerantSizeRatio)
```
Comment (lhy1024) [inline server/schedulers/utils.go:130]: can we move them to `adjustTolerantRatio`？

Comment (disksing) [reply server/schedulers/utils.go:130]: The `regionSize` is related to the region which is selected to balance so it cannot be calculated ahead.

**Review by rleungx — APPROVED** [review]

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**